cypress/support/commands.js

```diff
@@ -0,0 +1,39 @@
+function normalizeCoords(coords) {
+  // Note: Cypress seems to check against falsiness (thus 0 values are
+  //  considered as not supplied, and position defualts to "center").
+  //  Until fixed, default to `1`
+  // See https://github.com/cypress-io/cypress/issues/2338
+  return {
+    x: coords.x || 1,
+    y: coords.y || 1
+  };
+}
+
+Cypress.Commands.add("pickTool", { prevSubject: false }, tool =>
+  cy.contains("label", tool).click()
+);
+
+Cypress.Commands.add(
+  "canvasClick",
+  { prevSubject: true },
+  ($canvas, coords) => {
+    coords = normalizeCoords(coords);
+    cy.wrap($canvas)
+      .trigger("mousedown", coords.x, coords.y)
+      .trigger("mouseup", coords.x, coords.y);
+  }
+);
+
+Cypress.Commands.add(
+  "canvasDrag",
```

Comment:
Instead of `canvasDrag`, `drag` is more appropriate because is more generic.
Instead of passing the `startCoords` and `endCoords` as objects, you can pass two arrays, we will not have other keys.

Reply:
I wanted the API to be more explicit (`[50, 50], [100, 100]`) may not be quite obvious, especially if we don't use TS in tests at the moment.

Reply:
I want to keep it consistent with cypress signature https://docs.cypress.io/api/commands/click.html#Syntax